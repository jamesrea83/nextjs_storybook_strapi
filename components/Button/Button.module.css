.button {
	all: unset;
	display: flex;
	align-items: center;
	justify-content: center;
	user-select: none;
	cursor: pointer;
	font-size: 1.6rem;
	width: 15rem;
	height: 4rem;
	border-radius: 1rem;
	transition: all 0.4s ease;
	box-shadow: 0.5vmin 0.5vmin 1vmin #c8d0e7, -0.5vmin -0.5vmin 1vmin #ffffff;
}

.button:hover {
	opacity: 0.9;
}

.button:active {
	box-shadow: 0.5vmin 0.5vmin 1vmin #c8d0e7 inset,
		-0.5vmin -0.5vmin 1vmin #ffffff inset;
}

.primary {
	background: lightseagreen;
	color: white;
}

.secondary {
	background: lightskyblue;
	color: white;
}

.danger {
	background: #c01414;
	color: white;
}

.warning {
	background: orangered;
	color: white;
}
