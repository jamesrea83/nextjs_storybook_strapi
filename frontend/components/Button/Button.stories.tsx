import { Meta, StoryObj } from '@storybook/react';
import { expect, jest } from '@storybook/jest';
import { Button } from '.';
import { screen, userEvent } from '@storybook/testing-library';

export default {
	title: 'Controls/Button',
	component: Button,
	args: {
		children: 'Button',
		onClick: jest.fn(),
	},
} as Meta<typeof Button>;

type Story = StoryObj<typeof Button>;

export const Primary: Story = {
	play: async ({ args }) => {
		await userEvent.click(screen.getByRole('button'));
		await expect(args.onClick).toHaveBeenCalled();
	},
	args: {
		color: 'primary',
	},
};

export const Secondary: Story = {
	...Primary,
	args: {
		color: 'secondary',
	},
};

export const Warning: Story = {
	...Primary,
	args: {
		color: 'warning',
	},
};

export const Danger: Story = {
	...Primary,
	args: {
		color: 'danger',
	},
};
