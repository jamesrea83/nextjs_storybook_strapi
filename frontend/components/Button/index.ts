export {
	Button,
	PrimaryButton,
	SecondaryButton,
	DangerButton,
	WarningButton,
} from './Button';
