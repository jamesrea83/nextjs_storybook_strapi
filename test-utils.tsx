import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';

import { Themes } from './styles/themes';

interface Props {
	children: ReactNode;
}

const Wrapper = ({ children }: Props) => (
	<ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
	render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
