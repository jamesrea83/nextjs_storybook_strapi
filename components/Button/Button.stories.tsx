import * as React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import Button from './Button';

export default {
	title: 'Custom/Button',
	component: Button,
} as Meta;

export const Primary: StoryFn = args => <Button {...args} />;
Primary.args = {
	label: 'Custom/Button',
	primary: true,
};
