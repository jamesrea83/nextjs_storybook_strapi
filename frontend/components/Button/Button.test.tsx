import { render, screen } from '@/test-utils';
import userEvent from '@testing-library/user-event';
import { Props } from './Button';
import { Button } from '.';

describe('Button tests', () => {
	it('should render', () => {
		const buttonProps: Props = {
			children: 'Test Button',
			color: 'primary',
			onClick: jest.fn(),
		};
		const { asFragment } = render(<Button {...buttonProps} />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should call onClick on click', async () => {
		const buttonProps: Props = {
			children: 'Test Button',
			color: 'primary',
			onClick: jest.fn(),
		};

		render(<Button {...buttonProps} />);

		const element = screen.getByRole('button');
		await userEvent.click(element);

		expect(buttonProps.onClick).toHaveBeenCalled();
	});
});
