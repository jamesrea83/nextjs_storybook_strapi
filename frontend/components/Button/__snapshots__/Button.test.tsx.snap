// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should render 1`] = `
<DocumentFragment>
  <button
    class="css-odwgcq"
    color="primary"
  >
    Test Button
  </button>
</DocumentFragment>
`;
